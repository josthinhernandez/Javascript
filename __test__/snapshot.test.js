import { getCharacter } from "../snapshot";

import rick from "../rick.json";

describe('Es hora de las instantaneas', () => {
    test("Snapshot", () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    test("Siempre fallara la instantanea", () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20),
        }
        expect(user).toMatchSnapshot();
    })

    test("Tenemos una excepcion dentro de codigo", () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: "Josthin Hernandez"
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        })
    });
})