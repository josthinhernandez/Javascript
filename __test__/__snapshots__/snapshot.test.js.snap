// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2021-07-05T17:53:04.837Z,
  "id": 12,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Female",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excepcion dentro de codigo 1`] = `
Object {
  "id": Any<Number>,
  "name": "Josthin Hernandez",
}
`;
